packages/@headlessui-react/src/hooks/use-server-handoff-complete.ts

```diff
@@ -1,8 +1,43 @@
-import { useState, useEffect } from 'react'
+import * as React from 'react'
 import { env } from '../utils/env'
 
+/**
+ * This is used to determine if we're hydrating in React 18.
+ *
+ * The `useServerHandoffComplete` hook doesn't work with `<Suspense>`
+ * because it assumes all hydration happens at one time during page load.
+ *
+ * Given that the problem only exists in React 18 we can rely
+ * on newer APIs to determine if hydration is happening.
+ *
+ * @returns
```

Comment:
We can probably drop this since we are using TypeScript already?

Reply:
Updated